subsys/net/lib/config/ieee802154_settings.c

```diff
@@ -17,23 +17,14 @@ LOG_MODULE_DECLARE(net_config, CONFIG_NET_CONFIG_LOG_LEVEL);
 #include <zephyr/net/net_mgmt.h>
 #include <zephyr/net/ieee802154_mgmt.h>
 
-int z_net_config_ieee802154_setup(struct net_if *iface)
+int z_net_config_ieee802154_setup(struct net_if *iface,
+				  uint16_t channel,
```

Comment:
Wouldn't it be nicer and more consistent with security settings if you passed in the settings struct?

Reply:
> Wouldn't it be nicer and more consistent with security settings if you passed in the settings struct?

We can tweak these things in following PRs.